File: common/src/main/java/generations/gg/generations/core/generationscore/world/level/block/entities/BallDisplayBlockEntity.java
package generations.gg.generations.core.generationscore.world.level.block.entities;

import generations.gg.generations.core.generationscore.client.model.ModelContextProviders;
import net.minecraft.core.BlockPos;
import net.minecraft.world.level.block.state.BlockState;

public class BallDisplayBlockEntity extends ModelProvidingBlockEntity implements ModelContextProviders.VariantProvider {
    public BallDisplayBlockEntity(BlockPos pos, BlockState state) {
        super(GenerationsBlockEntities.BALL_DISPLAY.get(), pos, state);
    }

    @Override
    public String getVariant() {
        return getBlockState().getBlock() instanceof BallDisplayBlock ballDisplayBlock ? ballDisplayBlock.getVariant() : "empty";
    }
}

File: src/common/main/kotlin/generations/gg/generations/core/generationscore/common/world/level/block/entities/BallDisplayBlockEntity.kt
package generations.gg.generations.core.generationscore.common.world.level.block.entities

import generations.gg.generations.core.generationscore.common.client.model.ModelContextProviders.VariantProvider
import generations.gg.generations.core.generationscore.common.client.render.rarecandy.instanceOrNull
import net.minecraft.core.BlockPos
import net.minecraft.world.level.block.state.BlockState

class BallDisplayBlockEntity(pos: BlockPos, state: BlockState) : ModelProvidingBlockEntity(GenerationsBlockEntities.BALL_DISPLAY, pos, state), VariantProvider {
    override fun getVariant(): String {
        return blockState.block.instanceOrNull<BallDisplayBlock>()?.variant ?: "empty"
    }
}
